extensions/stats/plugin.h

```diff
@@ -0,0 +1,443 @@
+    destination_principal = ctx.request.destination_principal;
+
+    auto destination_labels = ctx.destination.labels();
+
+    destination_app = destination_labels["app"];
+    destination_version = destination_labels["version"];
+
+    destination_service_host = ctx.request.destination_service_host;
+    destination_service_name = ctx.destination.workload_name();
+    destination_service_namespace = ctx.destination.namespace_();
+
+    request_protocol = ctx.request.request_protocol;
+    response_code = std::to_string(ctx.request.response_code);
+    response_flags =
+        ctx.request.response_flag.empty() ? vDash : ctx.request.response_flag;
+
+    connection_security_policy =
+        ctx.outbound ? unknown : (ctx.request.mTLS ? vMTLS : vNone);
+
+    setFieldsUnknownIfEmpty();
+
+    outbound = ctx.outbound;
+    mapped = true;
+  }
+
+  std::string to_string() const {
+#define DIMENSION(name) "\"", #name, "\":\"", name, "\" ,",
+    return absl::StrCat("{" ISTIO_DIMENSIONS "}");
+#undef DIMENSION
+  }
+
+  // debug function to specify a textual key.
+  // must match HashValue
+  std::string debug_key() {
+    auto key = absl::StrJoin({reporter, request_protocol, response_code,
+                              response_flags, connection_security_policy},
+                             "#");
+    if (outbound) {
+      return absl::StrJoin(
+          {key, destination_app, destination_version, destination_service_name,
+           destination_service_namespace},
+          "#");
+    } else {
+      return absl::StrJoin({key, source_app, source_version, source_workload,
+                            source_workload_namespace},
+                           "#");
+    }
+  }
+
+  // smart hash uses fields based on context.
+  // This function is required to make IstioDimensions type hashable.
+  template <typename H>
+  friend H AbslHashValue(H h, const IstioDimensions& c) {
+    h = H::combine(std::move(h), c.request_protocol, c.response_code,
+                   c.response_flags, c.connection_security_policy, c.outbound);
+
+    if (c.outbound) {  // only care about dest properties
+      return H::combine(std::move(h), c.destination_service_namespace,
+                        c.destination_service_name, c.destination_app,
+                        c.destination_version);
+    } else {  // only care about source properties
+      return H::combine(std::move(h), c.source_workload_namespace,
+                        c.source_workload, c.source_app, c.source_version);
+    }
+  }
+
+  // This function is required to make IstioDimensions type hashable.
+  friend bool operator==(const IstioDimensions& lhs,
+                         const IstioDimensions& rhs) {
+    return (
+#define DIMENSION(name) lhs.name == rhs.name&&
+        ISTIO_DIMENSIONS
+#undef DIMENSION
+            lhs.outbound == rhs.outbound);
+  }
+};
+
+// Node holds node_info proto and a computed key
+struct Node {
```

Comment:
The Node is not used any more, no? Did I miss its usage somewhere? If not used, then the key concat is also unnecessary right?